pkg/provider/terraform/instance/apply.go

```diff
@@ -136,6 +149,53 @@ type tfFuncs struct {
 	tfStateList func() (map[TResourceType]map[TResourceName]struct{}, error)
 }
 
+// hasRecentDeltas returns true if any tf.json[.new] files have been changed in
+// in the last "window" seconds
+func (p *plugin) hasRecentDeltas(window int) (bool, error) {
+	if err := p.fsLock.TryLock(); err != nil {
+		log.Infof("In handleFiles, cannot acquire file lock")
+		return false, err
+	}
+	defer p.fsLock.Unlock()
+
+	modTime := time.Time{}
+	fs := &afero.Afero{Fs: p.fs}
+	err := fs.Walk(p.Dir,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				log.Debugf("Ignoring file %s due to error: %s", path, err)
+				return nil
+			}
+			if m := tfFileRegex.FindStringSubmatch(info.Name()); len(m) == 3 {
+				if modTime.Before(info.ModTime()) {
+					modTime = info.ModTime()
+				}
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		return false, err
+	}
+	if !modTime.IsZero() {
+		now := time.Now()
```

Comment:
This is really a clever idea.  Essentially we are introducing the idea of check points into the processing of tf config files.

I am wondering here with this, will we be missing updates when the plugin shuts down and comes back up?  What if we write the timestamp of last apply and use that as a check for all the modifications since that checkpoint?